### ISSUE TITLE
Kobo Auro H2O Edition 2 v2 Touchscreen Problems

### ISSUE THREAD

**Issue by StefanOnGit** [koreader/koreader#3925]
* KOReader version: koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1654-g666c9f1_2018-04-12
* Device: Kobo Auro H2O Edition 2

#### Issue
If koreader is started with Kobo Start Menu (over the webinterface since the KSM guys have no drivers for the touchscreen) it does not respond to touch (sometimes it responds but at the wrong position).

#### Steps to reproduce
Connect Device to Wifi
Install Kobo Start Menu
Reboot
When Device shows the Kobo Start Menu login in the webinterface (ip:8087/rmc_test.html) and launch "start koreader"
Result: Koreader gets started and does not respond to touch correctly.

##### `crash.log` (if applicable)
It does not Crash

**Comment by NiLuJe** [comment]
H2O2 v1 or v2? On Which FW version?

Does it work when launched through a hardware-agnostic solution (i.e., fmon/KFMon)?

**Comment by NiLuJe** [comment]
c.f., https://www.mobileread.com/forums/showthread.php?t=296022 for details on how to tell apart a v1 from a v2.

**Comment by NiLuJe** [comment]
In any case, I think we have some code to dump & debug the relevant info when the touch protocol is acting wonky, but I'm not familiar with it, so I'll let others enlighten you no that front ;).

**Comment by StefanOnGit** [comment]
It's a Kobo Aura H20 Edition 2 with Softwareversion 4.6.10197 (that is version 2 I think)
Same issue with Kfmon.
Didn't try it with fmon since outdated.

**Comment by Frenzie** [comment]
@NiLuJe It seems to be an H2O2v2. This issue is an offshoot from https://www.mobileread.com/forums/showthread.php?p=3689372#post3689372

@StefanOnGit What's the code name, btw? The simplest course of action for you is to add a definition and try a few combinations of existing protocols like I described. Should that yield no result I've described some follow up steps here: https://www.mobileread.com/forums/showthread.php?p=3609611#post3609611

(I don't understand where all these stupid custom protocols keep coming from, but hopefully that's not what it is.)

**Comment by NiLuJe** [comment]
Unfortunately I think it's *still* codenamed snow, so that's going to be fun...

**Comment by Frenzie** [comment]
Hm, that's too bad. In any case @StefanOnGit the obvious place to start is just some quick tests with the definition, regardless which one you use for testing, see here:

https://github.com/koreader/koreader/blob/43d09e245d29b8c490d4b6f9abef80640c7bb200/frontend/device/kobo/device.lua#L108-L110

First remove those lines for the properly standard kernel-based touch protocol, then try `touch_phoenix_protocol`, etc. The idea being it'd only take a couple of minutes to do these basic checks without doing anything complex or time-wasting. You can pretty much just keep the file open in SSH or in an FTP program like FileZilla over SSH and cycle through quickly.

**Comment by Frenzie** [comment]
But be very sure you're actually using whichever definition you think you are! Otherwise you're testing nothing at all. ;-)

**Comment by cramoisi** [comment]
https://www.mobileread.com/forums/showthread.php?t=296648

**Comment by NiLuJe** [comment]
I've implemented the required fiddling with product code to ID it properly, this'll make it in my next PR.

This will *NOT* fix it, simply make it easy to make it work without breaking anything else.

**Comment by cairnsh** [comment]
I think it works for me (with a h2o2v2) to delete the lines
```touch_snow_protocol = true,
touch_mirrored_x = false,```
from device.lua. Apparently it's just using the normal multitouch protocol.

**Comment by Frenzie** [comment]
Also without touch_probe_ev_epoch_time?

**Comment by cairnsh** [comment]
No, I left that in. (Although if I remember correctly, that is only a probe and won't do anything if it's not needed?)

**Comment by NiLuJe** [comment]
@cairnsh : Just to be perfectly clear, that's:

```
model = "Kobo_snow",
    hasFrontlight = yes,
    touch_probe_ev_epoch_time = true,
    display_dpi = 265,
```

Right?

Like @Frenzie said, if you could test without `touch_probe_ev_epoch_time`, that'd be greatly appreciated ;).

**Comment by cantelope** [comment]
I believe I have a H2O2v2.  F/W version 4.8.11073, and 7th char of the serial number is a 2.  I've tried installing koreader using the KFMon only.

koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1735-g14e7730_2018-06-02.zip:
- touch screen is totally unresponsive, and I have to reboot
- Added the commit 6a04dd8dc60572904a50258f3d8052864c0d2805 (i.e. removed the touch_phoenix_protocol = true line from the KoboSnowRev2 definition), but the touch screen is still unresponsive
- but I edited the device.lua file by converting to dos line endings using wordpad.  is that ok?

koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1654-g666c9f1_2018-04-12.zip
- touch screen seems to work, but my touches seem to be flipped horizontally. i.e. to tap on a button on the left of the screen, I have to physically tap on the right of the screen.

**Comment by NiLuJe** [comment]
@cantelope : Huh. What about with the current [nightly](http://ota.koreader.rocks/koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1743-g6e8cb23_2018-06-14.targz)?

**Comment by NiLuJe** [comment]
I just realized that I messed up the Rev2 detection for people running KSM, this will be fixed shortly ;).

**Comment by cantelope** [comment]
Aha - touch on that nightly http://ota.koreader.rocks/koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1743-g6e8cb23_2018-06-14.targz works without tweaks, as it should if it's the same.

**Comment by StefanOnGit** [comment]
Hi all, im back - now in learning phase so I had some time to get the reader finally working.

What I did:
- Reset the device
- Install KFmon
- Install the koreader [nightly build](http://ota.koreader.rocks/koreader-kobo-arm-kobo-linux-gnueabihf-v2015.11-1743-g6e8cb23_2018-06-14.targz)

When I restarted the device and tapped on the koreader.png the koreader shows up, but touch does not response. I've restarted the device again and tapped on the koreader.png. Now koreader shows the file browser, but the touch does still not work.

So I took a look at my software version that was something with 4.6 so I did:
- Downloaded and installed [latest firmware update 4.8.XXX](https://geek1011.github.io/KoboStuff/kobofirmware.html)

Now kfmon was broken, therefore:
- Copied content of the same kfmon zip into the kobo root (overwriting stuff)
- Let the device install it

And now it works! I love reading my scientific pdfs in the sun at the lake, thank you all for making this possible 👍 

Btw: should i close this thread?
